File: touyou/GitMeAndroid/app/src/main/java/com/dev/touyou/gitme/Data/Payload.kt
package com.dev.touyou.gitme.Data

import java.net.URL
import java.util.*

/**
 * Created by touyou on 2018/02/25.
 */

data class Payload(
        val action: String?,
        val comment: Comment?,
        val issue: Issue?,
        val before: String?,
        val commits: Array<Commit>?,
        val distinctSize: Int?,
        val head: String?,
        val pushId: Int?,
        val ref: String?,
        val refType: String?,
        val size: Int?,
        val description: String?,
        val masterBranch: String?,
        val pusherType: String?,
        val forkee: Forkee?
)

data class Comment(
        val authorAssociation: String,
        val body: String,
        val createdAt: Date,
        val htmlUrl: URL,
        val id: Int,
        val issueUrl: URL,
        val updatedAt: Date,
        val url: URL,
        val user: User
)

data class Issue(
        val assignee: String?,
        val assignees: Array<String>,
        val authorAssociation: String?,
        val body: String,
        val closedAt: Date?,
        val commentsCount: Int,
        val commentsUrl: URL,
        val createdAt: Date,
        val eventsUrl: URL,
        val htmlUrl: URL,
        val id: Int,
        val labels: Array<Label>,
        val labelsUrl: String,
        val lockedCount: Int,
        val milestone: String?,
        val number: Int,
        val repositoryUrl: URL,
        val state: String,
        val title: String,
        val updatedAt: Date,
        val url: String,
        val user: User
)

data class Author(
        val email: String,
        val name: String
)

data class Commit(
        val author: Author,
        val distinct: Int,
        val message: String,
        val sha: String,
        val url: URL
)

data class Label(
        val color: String,
        val default: Int,
        val id: Int,
        val name: String,
        val url: URL
)

data class Forkee()

File: touyou/GitMe/GitMe/Data/Payload.swift
//
//  Payload.swift
//  GitMe
//
//  Created by 藤井陽介 on 2018/01/31.
//  Copyright © 2018 touyou. All rights reserved.
//

import UIKit

// MARK: - Payload

struct Payload: Codable {

    /// IssueCommnet, Issues, Watch, CommitComment
    let action: String?
    /// IssueComment, CommitComment
    let comment: Comment?
    /// Push, Create, Delete
    let ref: String?
    /// Create, Delete
    let refType: String?
    /// Fork
    let forkee: Forkee?
    /// IssueComment, Issues
    let issue: Issue?
    /// Member
    let member: User?
    /// OrgBlock
    let blockedUser: User?
    /// PullRequestReview
    let pullRequest: PullRequest?
    /// Release
    let release: Release?

    enum CodingKeys: String, CodingKey {
        case action
        case comment
        case ref
        case refType = "ref_type"
        case forkee
        case issue
        case member
        case blockedUser = "blocked_user"
        case pullRequest = "pull_request"
        case release
    }
}

// MARK: - Comment

struct Comment: Codable {

    let body: String
    let createdAt: Date
    let updatedAt: Date
    let url: URL
    let user: User

    enum CodingKeys: String, CodingKey {

        case body
        case createdAt = "created_at"
        case updatedAt = "updated_at"
        case url
        case user
    }
}

// MARK: - Issue

struct Issue: Codable {

    let body: String
    let number: Int
    let repositoryUrl: URL
    let state: String
    let title: String
    let updatedAt: Date
    let url: String
    let user: User

    enum CodingKeys: String, CodingKey {

        case body
        case number
        case repositoryUrl = "repository_url"
        case state
        case title
        case updatedAt = "updated_at"
        case url
        case user
    }
}

// MARK: - Forkee

struct Forkee: Codable {

    let fullName: String?

    enum CodingKeys: String, CodingKey {
        case fullName = "full_name"
    }
}

// MARK: - PullRequest

struct PullRequest: Codable {

    let number: Int
}

// MARK: - Release

struct Release: Codable {

    let tagName: String

    enum CodingKeys: String, CodingKey {
        case tagName = "tag_name"
    }
}
